### DISCUSSION TITLE
/logout route as POST?

### DISCUSSION THREAD

**Discussion by Epzillon** [JakyeRU/Larascord#53] [Q&A] [answered]
How is the /logout route provided expected to be used? Using an `<a>` element with `href="{{ url('/logout') }}"` throws an error `The GET method is not supported for route logout. Supported methods: POST.`.

Are you intended to create a new route and POST from there?

**Comment by ValentinNeaga** [comment, marked as answer]
Hello @Epzillon,

The reason why the `/logout` route doesn't support the `GET` method by default is because logging out of an application involves modifying the user's session data, which is considered a state-changing operation. 

According to the HTTP specification, `GET` requests are intended to be used for retrieving resources without modifying their state, while POST requests are intended for state-changing operations.

As browsers can [prefetch](https://en.wikipedia.org/wiki/Link_prefetching) URLs, users may accidentally be logged out if the `GET` method is used.

By only supporting `POST` requests, the `/logout` route ensures that logging out of the application can only be performed through a state-changing operation, which helps to prevent accidental or malicious logout requests triggered by following a link or accessing a URL in the browser. 

Additionally, by using the `POST` method, it is possible to include sensitive data (such as session tokens) in the request body, which is not possible with the `GET` method.

Here is an example of how you can log out a user with a link through the `POST` method:
```html
<form method="POST" action="{{ route('logout') }}">
    @csrf

    <a href="{{ route('logout') }}" onclick="event.preventDefault(); this.closest('form').submit();">
        {{ __('Log Out') }}
    </a>
</form>
```

You can also check [Logout: GET or POST?](https://stackoverflow.com/questions/3521290) for more information on this topic.

**Reply by Epzillon** [reply]
Thanks for replying so quickly! I assumed it would be something along those lines. I just didn't know how to implement it. Thanks for the thorough explanation and examples. I will attempt to use this solution 👍
